integration_tests/test_api.py

```diff
@@ -347,14 +347,21 @@ async def test_list_gridpool_trades(set_up: dict[str, Any]) -> None:
 
 
 @pytest.mark.asyncio
-async def test_list_public_trades(set_up: dict[str, Any]) -> None:
+async def test_list_public_trades_pages(set_up: dict[str, Any]) -> None:
     """Test listing public trades."""
-    public_trades = await set_up["client"].list_public_trades(
+    public_trades = []
+    page_limit = 3  # Limit the test to 3 pages
```

Comment:
If we only assert for len > 0, then we don't need to wait for three pages in, no?

Reply:
Actually, we even allow len >= 0, what exactly are we testing here :)